test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,244 @@
+            const accounts = await getIexecAccounts();
+            ({ iexecAdmin, anyone: anyoneA, requester: anyoneB, anyone } = accounts);
+
+            iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, iexecAdmin);
+            iexecPocoAsAnyone = iexecPoco.connect(anyone);
+        }
+
+        describe('Receive and Fallback', () => {
+            it('Should call receive successfully', async () => {
+                await expect(
+                    await anyoneA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: etherDepositAmount,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [anyoneA, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA], [depositAmount])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, depositAmount);
+            });
+
+            it('Should call fallback successfully', async () => {
+                const randomData = ethers.utils.hexlify(
+                    ethers.utils.toUtf8Bytes((Math.random() * 0xfffff).toString(16)),
+                );
+
+                await expect(
+                    await anyoneA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: etherDepositAmount,
+                        data: randomData,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [anyoneA, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA], [depositAmount])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, depositAmount);
+            });
+        });
+
+        describe('Deposits', () => {
+            it('Should deposit native tokens', async () => {
+                expect(await iexecPocoAsAnyone.callStatic.deposit(...depositArgs)).to.be.true;
+                await expect(iexecPocoAsAnyone.deposit(...depositArgs))
+                    .to.changeEtherBalances(
+                        [anyoneA, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA], [depositAmount])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, depositAmount);
+            });
+
+            it('Should deposit native tokens for another address', async () => {
+                const depositForArgs = [anyoneA.address, { value: etherDepositAmount }] as [
+                    string,
+                    { value: BigNumber },
+                ];
```

Comment:
Maybe it makes sense to declare args types (of deposit, depositFor,..) once in top (if used multiple times) and use them here

Reply:
Done